File: event-management/src/main/java/com/ikt/event/management/service/UserService.java
package com.ikt.event.management.service;

import com.ikt.event.management.entity.Role;
import com.ikt.event.management.entity.User;

import java.util.List;

public interface UserService {

    List<User> listAll();

    User findById(Integer personID);

    User create(String personName, String personEmail, String phoneNumber, String companyName, Role role);

    void personRegistrationForEvent(Integer userID, Integer roleID, Integer eventID, Role role);
}

File: src/services/userService.js
import axios from '../api/axios';

const UserService = {
    getUsers: () => {
        return axios.get("/users/all");
    },
    addUser: (personName, personEmail, phoneNumber, companyName) => {
        return axios.post("/users/create", {
            "personName": personName,
            "personEmail": personEmail,
            "phoneNumber": phoneNumber,
            "companyName": companyName
        });
    },
    getUser: (userID) => {
        return axios.get(`/users/${userID}`);
    },
    registerForEvent: (userID, roleId, eventId) => {
        return axios.post(`/users/${userID}/register`, {
            "roleId": roleId,
            "eventId": eventId
        });
    }
}

export default UserService;